clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -118,6 +130,24 @@ public void assign(List<TopicPartition> partitions) {
         this.assignment.keySet().retainAll(this.userAssignment);
     }
 
+    public void subscribe(Pattern pattern) {
+        if (!this.subscription.isEmpty() || !this.userAssignment.isEmpty())
```

Comment:
Haha, the corollary would be that we wouldn't need unsubscribe(pattern) anymore (we got rid of unsubscribe(topics) already). My preference is probably to override, but I feel a little concern that it would be confusing for users who expect to be able to do both (the exception makes the restriction very clear). I think another option we talked about was just to have a single unsubscribe() with no arguments. Maybe we can drag @gwenshap in here for another opinion since Guozhang is on vacation.